tests/integration/extenders/ApiControllerTest.php

```diff
@@ -0,0 +1,582 @@
+    }
+
+    /**
+     * @test
+     */
+    public function custom_limit_works_if_set()
+    {
+        $this->extend(
+            (new Extend\ApiController(ListDiscussionsController::class))
+                ->setLimit(1)
+        );
+
+        $this->prepDb();
+
+        $response = $this->send(
+            $this->request('GET', '/api/discussions', [
+                'authenticatedAs' => 1,
+            ])
+        );
+
+        $payload = json_decode($response->getBody(), true);
+
+        $this->assertCount(1, $payload['data']);
+    }
+
+    /**
+     * @test
+     */
+    public function custom_max_limit_works_if_set()
+    {
+        $this->extend(
+            (new Extend\ApiController(ListDiscussionsController::class))
+                ->setMaxLimit(1)
+        );
+
+        $this->prepDb();
+
+        $response = $this->send(
+            $this->request('GET', '/api/discussions', [
+                'authenticatedAs' => 1,
+            ])->withQueryParams([
+                'page' => ['limit' => '5'],
+            ])
+        );
+
+        $payload = json_decode($response->getBody(), true);
+
+        $this->assertCount(1, $payload['data']);
+    }
+
+    /**
+     * @test
+     */
+    public function custom_sort_field_doesnt_exist_by_default()
+    {
+        $this->prepDb();
+
+        $response = $this->send(
+            $this->request('GET', '/api/discussions', [
+                'authenticatedAs' => 1,
+            ])->withQueryParams([
+                'sort' => 'userId',
+            ])
+        );
+
+        $this->assertEquals(400, $response->getStatusCode());
+    }
+
+    /**
+     * @test
+     */
+    public function custom_sort_field_doesnt_work_with_false_callback_return()
+    {
+        $this->extend(
+            (new Extend\ApiController(ListDiscussionsController::class))
+                ->addSortField('userId', function () {
+                    return false;
+                })
+        );
+
+        $this->prepDb();
+
+        $response = $this->send(
+            $this->request('GET', '/api/discussions', [
+                'authenticatedAs' => 1,
+            ])->withQueryParams([
+                'sort' => 'userId',
+            ])
+        );
+
+        $this->assertEquals(400, $response->getStatusCode());
+    }
+
+    /**
+     * @test
+     */
+    public function custom_sort_field_exists_if_added()
+    {
+        $this->extend(
+            (new Extend\ApiController(ListDiscussionsController::class))
+                ->addSortField('userId')
+        );
+
+        $this->prepDb();
+
+        $response = $this->send(
+            $this->request('GET', '/api/discussions', [
+                'authenticatedAs' => 1,
+            ])->withQueryParams([
+                'sort' => 'userId',
+            ])
+        );
+
+        $payload = json_decode($response->getBody(), true);
+
+        $this->assertEquals(200, $response->getStatusCode());
+        $this->assertEquals(3, $payload['data'][0]['id']);
```

Comment:
Hmm can we pluck IDs and compare all IDs here? 

$this->assertEquals(EXPECTED_ORDER, Arr::pluck($payload['data'], 'id'));